libraries/AC_Avoidance/AP_OABendyRuler.h

```diff
@@ -23,6 +23,21 @@ class AP_OABendyRuler {
     // returns true and populates origin_new and destination_new if OA is required.  returns false if OA is not required
     bool update(const Location& current_loc, const Location& destination, const Vector2f &ground_speed_vec, Location &origin_new, Location &destination_new);
 
+    enum class OABendyType {
+        OA_BENDY_DISABLED   = 0,
+        OA_BENDY_HORIZONTAL = 1,
+        OA_BENDY_VERTICAL   = 2,
+    };
+
+    // Search for path in XY direction
+    bool search_xy_path(const Location& current_loc, const Location& destination, float ground_course_deg, Location &destination_new, float lookahead_step_1_dist, float lookahead_step_2_dist, float bearing_to_dest, float distance_to_dest);
+
+    // Search for path in the Vertical directions
+    bool search_vertical_path(const Location& current_loc, const Location& destination,Location &destination_new, const float &lookahead_step1_dist, const float &lookahead_step2_dist, const float &bearing_to_dest, const float &distance_to_dest); 
+
+    // return type of BendyRuler in use
+    OABendyType get_type() const;
```

Comment:
maybe move the get_type() method to be right under the enum definition.  not a blocker of course.